libs/sdk-backend-tiger/src/convertors/fromBackend/MetadataConverter.ts

```diff
@@ -0,0 +1,202 @@
+// (C) 2019-2021 GoodData Corporation
+import {
+    JsonApiAttribute,
+    JsonApiAttributeDocument,
+    JsonApiAttributeList,
+    JsonApiAttributeWithLinks,
+    JsonApiFactWithLinks,
+    JsonApiLabel,
+    JsonApiLabelDocument,
+    JsonApiLabelWithLinks,
+    JsonApiLinkage,
+    JsonApiMetricWithLinks,
+} from "@gooddata/api-client-tiger";
+import keyBy from "lodash/keyBy";
+import { IAttributeDisplayFormMetadataObject, IAttributeMetadataObject } from "@gooddata/sdk-backend-spi";
+import {
+    IMetadataObjectBuilder,
+    newAttributeDisplayFormMetadataObject,
+    newAttributeMetadataObject,
+} from "@gooddata/sdk-backend-base";
+import { idRef } from "@gooddata/sdk-model";
+import invariant from "ts-invariant";
+
+export type MetadataObjectFromApi =
+    | JsonApiAttributeWithLinks
+    | JsonApiFactWithLinks
+    | JsonApiMetricWithLinks
+    | JsonApiLabelWithLinks;
+
+export const commonMetadataObjectModifications = <
+    TItem extends MetadataObjectFromApi,
+    T extends IMetadataObjectBuilder
+>(
+    item: TItem,
+) => (builder: T) =>
+    builder
+        .id(item.id)
+        .uri(item.links!.self)
+        .title(item.attributes?.title || "")
+        .description(item.attributes?.description || "");
+
+function createLabelMap(included: any[] | undefined): Record<string, JsonApiLabelWithLinks> {
+    if (!included) {
+        return {};
+    }
+
+    const labels: JsonApiLabel[] = included
+        .map((include) => {
+            if ((include as JsonApiLinkage).type !== "label") {
+                return null;
+            }
+
+            return include as JsonApiLabelWithLinks;
+        })
+        .filter((include): include is JsonApiLabelWithLinks => include !== null);
+
+    return keyBy(labels, (t) => t.id);
+}
+
+/**
+ * Converts all labels of this attribute. The map contains sideloaded label information
+ */
+function convertAttributeLabels(
+    attribute: JsonApiAttribute,
+    labelsMap: Record<string, JsonApiLabelWithLinks>,
+): IAttributeDisplayFormMetadataObject[] {
+    const labelsRefs = attribute.relationships?.labels?.data as JsonApiLinkage[];
+
+    return labelsRefs
+        .map((ref) => {
+            const label = labelsMap[ref.id];
+
+            if (!label) {
+                return;
```

Comment:
Nitpick: we should probably return `undefined` explicitly, since we test against it in the `filter` below.